resources/js/components/LinkableTrendMetric.vue

```diff
@@ -1,177 +1,149 @@
-      :url="this.card.url"
-    />
-  </template>
-
-  <script>
-  import map from 'lodash/map'
-  import { InteractsWithDates, MetricBehavior } from '@/mixins'
-  import { minimum } from '@/util'
-
-  export default {
-    name: 'LinkableTrendMetric',
-
-    // mixins: [InteractsWithDates, MetricBehavior],
-
-    props: {
-      card: {
-        type: Object,
-        required: true,
-      },
-
-      resourceName: {
-        type: String,
-        default: '',
-      },
-
-      resourceId: {
-        type: [Number, String],
-        default: '',
-      },
-
-      lens: {
-        type: String,
-        default: '',
-      },
-    },
-
-    data: () => ({
-      loading: true,
-      value: '',
-      data: [],
-      format: '(0[.]00a)',
-      prefix: '',
-      suffix: '',
-      suffixInflection: true,
-      selectedRangeKey: null,
-    }),
-
-    watch: {
-      resourceId() {
-        this.fetch()
-      },
+  <component :is="card.url ? 'BaseLinkableTrendMetric' : 'BaseTrendMetric'"
+    @selected="handleRangeSelected"
+    :title="card.name"
+    :help-text="card.helpText"
+    :help-width="card.helpWidth"
+    :value="value"
+    :chart-data="data"
+    :ranges="card.ranges"
+    :format="format"
+    :prefix="prefix"
+    :suffix="suffix"
+    :suffix-inflection="suffixInflection"
+    :selected-range-key="selectedRangeKey"
+    :loading="loading"
+    :url="card.url"
+  />
+</template>
+
+<script>
+import map from 'lodash/map'
+import { MetricBehavior } from 'laravel-nova'
+import InteractsWithDates from '@/mixins/InteractsWithDates'
+
+export default {
+  name: 'LinkableTrendMetric',
+
+  mixins: [InteractsWithDates, MetricBehavior],
+
+  data: () => ({
+    loading: true,
+    value: '',
+    data: [],
+    format: '(0[.]00a)',
+    prefix: '',
+    suffix: '',
+    suffixInflection: true,
+    selectedRangeKey: null,
+  }),
+
+  watch: {
+    resourceId() {
+      this.fetch()
     },
-
-    created() {
-      if (this.hasRanges) {
-        this.selectedRangeKey =
-          this.card.selectedRangeKey || this.card.ranges[0].value
-      }
-
+  },
+
+  created() {
+    if (this.hasRanges) {
+      this.selectedRangeKey =
+        this.card.selectedRangeKey || this.card.ranges[0].value
+    }
+
+    this.fetch()
+  },
+
+  mounted() {
+    if (this.card && this.card.refreshWhenFiltersChange === true) {
+      Nova.$on('filter-changed', this.fetch)
+      Nova.$on('filter-reset', this.fetch)
+    }
+  },
+
+  beforeUnmount() {
+    if (this.card && this.card.refreshWhenFiltersChange === true) {
+      Nova.$off('filter-changed', this.fetch)
+      Nova.$on('filter-reset', this.fetch)
+    }
+  },
+
+  methods: {
+    handleRangeSelected(key) {
+      this.selectedRangeKey = key
       this.fetch()
     },
 
-    mounted() {
-      if (this.card && this.card.refreshWhenFiltersChange === true) {
-        Nova.$on('filter-changed', this.fetch)
+    /**
+     * @returns [Function]
+     */
+    handleFetchCallback() {
+      return ({
```

Comment:
Avoid declaring `fetch()` but instead just handle the response using `handleFetchCallback()`